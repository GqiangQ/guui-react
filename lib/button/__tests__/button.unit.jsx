import {mount} from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../button'


describe(Button.displayName, () => {
  it('DisplayName', () => {
    expect(Button.displayName).toEqual('Button')
  })
  it('prop icon = alipay', () => {
    const tree = renderer
      .create(<Button icon="alipay"><span>Click Me</span></Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop iconPosition = right', () => {
    const tree = renderer
      .create(<Button icon="alipay" iconPosition="right">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop iconFill = red', () => {
    const tree = renderer
      .create(<Button icon="alipay" iconFill="red">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop size = small', () => {
    const tree = renderer
      .create(<Button size="small">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop size = big', () => {
    const tree = renderer
      .create(<Button size="big">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop level = import', () => {
    const tree = renderer
      .create(<Button level="important">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop level = danger', () => {
    const tree = renderer
      .create(<Button level="danger">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop ghost = true', () => {
    const tree = renderer
      .create(<Button ghost={true}>Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop badge = 42', () => {
    const tree = renderer
      .create(<Button badge={42}>Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop href = http://qq.com', () => {
    const tree = renderer
      .create(<Button href="http://qq.com">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop target = x', () => {
    const tree = renderer
      .create(<Button href="http://qq.com" target="x">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop disabled = true', () => {
    const fn = jest.fn()
    const c = mount(<Button onClick={fn} disabled={true}>Click Me</Button>)
    c.find('button').simulate('click')
    expect(fn.mock.calls.length).toEqual(0)
  })
  it('prop loading = true', () => {
    const tree = renderer
      .create(<Button loading={true}>Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('prop type = submit', () => {
    const tree = renderer
      .create(<Button type="submit">Click Me</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should response to click event', () => {
    const fn = jest.fn()
    const c = mount(<Button onClick={fn}>Click Me</Button>)
    c.find('button').simulate('click')
    expect(fn.mock.calls.length).toEqual(1)
  })
  it('should prevent disabled link', () => {
    const fn = jest.fn()
    const c = mount(<Button href="http://qq.com" disabled={true}>Click Me</Button>)
    c.find('a').simulate('click')
    expect(fn.mock.calls.length).toEqual(0)
  })
  it('should switch icon when loading turn to true', () => {
    const fn = jest.fn()
    const c = mount(<Button icon="alipay" loading={false}>Click Me</Button>)
    expect(c.find('use').getDOMNode().getAttribute('xlink:href')).toEqual('#alipay')
    c.setProps({loading: true})
    expect(c.find('use').getDOMNode().getAttribute('xlink:href')).toEqual('#loading')
  })
  it('should animate when loading turn to true', () => {
    const fn = jest.fn()
    const c = mount(<Button loading={false}>Click Me</Button>)
    expect(c.find('use').exists()).toEqual(false)
    c.setProps({loading: true})
    expect(c.find('use').getDOMNode().getAttribute('xlink:href')).toEqual('#loading')
  })
})
