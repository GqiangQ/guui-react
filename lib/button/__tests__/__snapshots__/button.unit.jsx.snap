// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button prop badge = 42 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
  <div
    className="gu-button-badge"
  >
    <span>
      42
    </span>
  </div>
   
</button>
`;

exports[`Button prop ghost = true 1`] = `
<button
  className="gu-button gu-button-default gu-button-ghost "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop href = http://qq.com 1`] = `
<a
  className="gu-button gu-button-default "
  href="http://qq.com"
  onClick={[Function]}
>
   
   
  <span>
    Click Me
  </span>
   
   
</a>
`;

exports[`Button prop icon = alipay 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  <svg
    className="gu-icon gu-icon-alipay "
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#alipay"
    />
  </svg>
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop iconFill = red 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  <svg
    className="gu-icon gu-icon-alipay "
    style={
      Object {
        "fill": "red",
      }
    }
  >
    <use
      xlinkHref="#alipay"
    />
  </svg>
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop iconPosition = right 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
  <span>
    Click Me
  </span>
   
  <svg
    className="gu-icon gu-icon-alipay "
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#alipay"
    />
  </svg>
   
   
</button>
`;

exports[`Button prop level = danger 1`] = `
<button
  className="gu-button gu-button-danger "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop level = import 1`] = `
<button
  className="gu-button gu-button-important "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop loading = true 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={true}
  onClick={[Function]}
  type="button"
>
   
  <svg
    className="gu-icon gu-icon-loading gu-button-loadingIcon"
    style={
      Object {
        "fill": undefined,
      }
    }
  >
    <use
      xlinkHref="#loading"
    />
  </svg>
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop size = big 1`] = `
<button
  className="gu-button gu-button-default gu-button-big "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop size = small 1`] = `
<button
  className="gu-button gu-button-default gu-button-small "
  disabled={false}
  onClick={[Function]}
  type="button"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;

exports[`Button prop target = x 1`] = `
<a
  className="gu-button gu-button-default "
  href="http://qq.com"
  onClick={[Function]}
  target="x"
>
   
   
  <span>
    Click Me
  </span>
   
   
</a>
`;

exports[`Button prop type = submit 1`] = `
<button
  className="gu-button gu-button-default "
  disabled={false}
  onClick={[Function]}
  type="submit"
>
   
   
  <span>
    Click Me
  </span>
   
   
</button>
`;
